src/compiler/crystal/codegen/target.cr

```diff
@@ -171,7 +171,16 @@ class Crystal::Codegen::Target
       raise Target::Error.new("Unsupported architecture for target triple: #{self}")
     end
 
-    opt_level = release ? LLVM::CodeGenOptLevel::Aggressive : LLVM::CodeGenOptLevel::None
+    opt_level = case optimization_mode
+                when Compiler::OptimizationMode::O3
+                  LLVM::CodeGenOptLevel::Aggressive
+                when Compiler::OptimizationMode::O2
+                  LLVM::CodeGenOptLevel::Default
+                when Compiler::OptimizationMode::O1
+                  LLVM::CodeGenOptLevel::Less
+                else
+                  LLVM::CodeGenOptLevel::None
+                end
```

Comment:
**suggestion:** Exhaustive case with predicate methods is type safe and more concise:
```suggestion
    opt_level = case optimization_mode
                in .o3? then LLVM::CodeGenOptLevel::Aggressive
                in .o2? then LLVM::CodeGenOptLevel::Default
                in .o1? then LLVM::CodeGenOptLevel::Less
                in .o0? then LLVM::CodeGenOptLevel::None
                end
```